client/src/main/scripts/slipstream.bootstrap.py

```diff
@@ -381,29 +403,80 @@ def _add_executor_to_initd(executor_name):
         cmd = 'update-rc.d %s defaults' % service_name
     else:
         cmd = 'chkconfig --add %s' % service_name
-    commands.getstatusoutput(cmd)
+
+    rc, output = _get_rc_output(cmd)
+    if rc != 0:
+        raise Exception('Failed registering machine executor with initd: %s' % output)
 
     return service_name
 
 
+def _configure_systemd_service(executor_name):
+    """
+    :param executor_name: name of the executor (node or orchestrator)
+    :return: systemd service name
+    """
+
+    service_name = _add_executor_to_systemd(executor_name)
+
+    return service_name
+
+
+def _add_executor_to_systemd(executor_name):
+    sname = 'slipstream-%s.service' % executor_name
+
+    src = '/opt/slipstream/client/etc/%s' % sname
+    dst = '/etc/systemd/system/%s' % sname
+
+    try: os.unlink(dst)
+    except: pass
+    os.symlink(src, dst)
+
+    rc, output = _get_rc_output('systemctl daemon-reload')
+    if rc != 0:
+        raise Exception('Failed registering machine executor with systemd: %s' % output)
+
+    return sname
+
+
+def _get_rc_output(cmd):
+    status, output = commands.getstatusoutput(cmd)
+    return os.WEXITSTATUS(status), output
+
+
 def _create_executor_config(executor_name):
-    with open('/etc/default/slipstream-%s' % executor_name, 'w') as fh:
-        fh.write('export PYTHONPATH=$PYTHONPATH:%s\n' % os.path.join(SLIPSTREAM_CLIENT_HOME, 'lib'))
-        fh.write('export PATH=$PATH:%s\n' % os.path.join(SLIPSTREAM_CLIENT_HOME, 'bin'))
-        fh.write('export PATH=$PATH:%s\n' % os.path.join(SLIPSTREAM_CLIENT_HOME, 'sbin'))
-        fh.write('export DAEMON_ARGS="%s"\n' % _get_verbosity())
-        fh.write('export SLIPSTREAM_CONNECTOR_INSTANCE="%s"\n' % os.environ.get('SLIPSTREAM_CONNECTOR_INSTANCE'))
-        if executor_name == 'orchestrator':
-            cloud_name = os.environ['SLIPSTREAM_CLOUD']
-            fh.write('export PYTHONPATH=$PYTHONPATH:%s\n' % os.path.join(os.sep, 'opt', cloud_name.lower()))
-
-            env_matcher = re.compile('SLIPSTREAM_')
-            for var, val in os.environ.items():
-                if env_matcher.match(var) and var != 'SLIPSTREAM_NODE_INSTANCE_NAME':
-                    #if re.search(' ', val) and not (val.startswith('"') and val.endswith('"')):
-                    if re.search(' ', val) and not 'SLIPSTREAM_COOKIE':
```

Comment:
I'm afraid this was committed in by mistake.  That `not 'SLIPSTREAM_COOKIE'` doesn't make sense at all.